resolwe/flow/models/utils.py

```diff
@@ -747,3 +747,72 @@ def referenced_files(data, include_descriptor=True):
     )
 
     return special_files + output_files + descriptor_files
+
+
+def get_collection_of_input_entities(data):
+    """Get collection that contains all "entity inputs" of a given data.
+
+    With "entity input", one refers to the inputs that are part of an entity.
+    """
+    # Prevent circular imports:
+    from resolwe.flow.models import Data, Collection
+
+    collections = set()
+
+    for field_schema, fields in iterate_fields(data.input, data.process.input_schema):
+        name = field_schema["name"]
+        value = fields[name]
+        if "type" not in field_schema:
+            continue
+        if field_schema["type"].startswith("data:"):
+            if value is None:
+                continue
+            try:
+                data = Data.objects.get(id=value)
+                if data.entity and data.collection:
+                    collections.add(data.collection.id)
+            except Data.DoesNotExist:
+                # This exception is copy-pasted from hydrate_input_references
+                # TODO: Think what is the case when data cannot be found? How process
```

Comment:
I believe this could happen:
- job is created, all input data is there
- job is waiting in queue, somebody deletes input data for the job
- job is started and input data is missing. 

If this is the case then aborting the job would be the right thing to do.

Reply:
But this should than also be fixed in `hydrate_input_references`, right?

Reply:
Just to write this down: After some discussion, we agreed to just let this be as is. Will fix in the future if proven necessary.